tcms/settings/common.py

```diff
@@ -130,6 +130,7 @@
     'global_login_required.GlobalLoginRequiredMiddleware',
     'simple_history.middleware.HistoryRequestMiddleware',
     'tcms.core.middleware.CheckSettingsMiddleware',
+    'tcms.core.middleware.CheckUnappliedMigrationsMiddleware'
```

Comment:
Testing will need the migrations from Django applied b/c we're looking for Site & later User and if no migration are applied everything starts failing even before we get to the new middleware.

Reply:
I was confused about how you meant for @ skipUnless to be applied here so I tried out setting up a small test app that we could use test with and without unapplied migrations by overriding the settings. I'm thinking this could work?

Reply:
It will work but it is not what we want to do here. The existing test suite is extensive as it is, no need to complicate things even more.  With GitHub Actions you have the ability for custom setup and custom job definitions, with @skipUnless you can skip test classes/test methods based on conditions.

So you can have a new GH job definition which applies only some migartions as a bare minimum so the app doesn't crash; possibly define some ENV variable to indicate you want to run the tests about this particular middleware and then tell manage.py test to execute only tests for this middleware (b/c the rest will fail and are not of interest here).  You don't need to introduce a test app or dummy migrations.

Reply:
I think instead of introducing a new ENV var, Django's test runner lets you define the specific test you want to run via the command.

Reply:
I'm having issues with this. The test runner automatically applies migrations against the test database.

Reply:
> I think instead of introducing a new ENV var, Django's test runner lets you define the specific test you want to run via the command.

that's true, but when we run all of the other tests (e.g. with MariaDB, Postgres, SQLite) we don't specify anything to the test runner so it needs a way to know how to exclude/skip the special test for migrations middleware. The easiest way to do this with ENV variables. The same way it is done when running tests for the internal bug tracker functionality or for external bug trackers.

Reply:
>  I'm having issues with this. The test runner automatically applies migrations against the test database.

Well show me how you are starting your test job then? I don't see a separate test job in this PR yet.

Reply:
Sorry, hadn't pushed yet because I hadn't got to a solution yet. Will push